src/components/color_generator/index.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+
+class ColorGenerator extends Component {
+  constructor({ savePalette }) {
+    super({ savePalette })
+    this.state = {
+      id: 0,
+      palette_name: '',
+      color1: {hex: '', lockType: 'lock-open'},
+      color2: {hex: '', lockType: 'lock-open'},
+      color3: {hex: '', lockType: 'lock-open'},
+      color4: {hex: '', lockType: 'lock-open'},
+      color5: {hex: '', lockType: 'lock-open'}
+    }
+  }
+
+  handleName = (e) => {
+    const { name, value } = e.target;
+
+    this.setState({ [name]: value })
+  }
+
+  handleColor = (e) => {
+    const { name, value } = e.target;
+
+    let color = this.state[name];
+    color.hex = value;
+
+    this.setState({ [name]: color });
+  }
+
+  handleLock = (e) => {
+    const { id } = e.target;
+    let color = this.state[id];
+    console.log(id, color)
+    // const lock = 'lock';
+    // const unlock = 'lock-open';
+
+    if (color.lockType === 'lock') {
+      color.lockType = 'lock-open';
+    } else {
+      color.lockType = 'lock'
+    }
+
+    this.setState({ [id]: color })
+  }
+
+  render() {
+    let {palette_name, color1, color2, color3, color4, color5} = this.state;
+
+    console.log('CG_state:', this.state)
+    return (
+      <form className="palette-form">
+        <input
+          name="palette_name"
+          onChange={e => this.handleName(e)}
+          value={palette_name}
+          className="palette-name"
+          type="text"
+          placeholder="Palette Name"
+          maxLength="10"
+        />
+        <fieldset className="color-field">
```

Comment:
I think we can turn the five color fieldsets into a map to prevent repetition/ keep the code DRY.